SimG4CMS/Calo/src/HFShowerLibrary.cc

```diff
@@ -226,10 +205,11 @@ std::vector<HFShowerLibrary::Hit> HFShowerLibrary::fillHits(const G4ThreeVector
 
   std::vector<HFShowerLibrary::Hit> hit;
   ok = false;
-  if (parCode == pi0PDG || parCode == etaPDG || parCode == nuePDG ||
-      parCode == numuPDG || parCode == nutauPDG || parCode == anuePDG ||
-      parCode == anumuPDG || parCode == anutauPDG || parCode == geantinoPDG)
+  bool isEM = G4TrackToParticleID::isGammaElectronPositron(parCode);
+  // shower is built only for gamma, e+- and stable hadrons
+  if (!isEM && !G4TrackToParticleID::isStableHadron(parCode)) { 
```

Comment:
hi @civanch @fabiocos -following my questions from the release meeting - I would suggest to add some debug printout here to see in which cases these two stanzas of code are not equivalent in _practice_ for some generic sample. It looks like a few events are enough to know the common examples.

Reply:
Hi @davidlange6 , I practically did this when was trying to clean up calorimeter sensitive detectors. Now this PR #21964 is closed, because we decided first to fix a weak place - selection of particles for SL.  In #21984 I was doing exactly as you propose and achieve at the end all WF except 2 to be identical. Most part of problems come from Castor - secondary particles hit it even for typical ttbar events (primary are filtered out by eta cut) and for a hit random numbers are used. Also association between hits and MCtruth is sensitive to minor details of HF and Castor. What I also observed : even if random sequence is identical there are differences in results if hits are not absolutely identical.

For this PR I do not do the same but it is clear, that depending of WF it is various hyperons, which are now do not considered for shower libraries. I do not see a need to include them in the list, because they can safely decay and the products of their decay will be treated by SL. Unfortunately, for sampling of a hyperon decay random numbers are also spent.

Reply:
Right - I don't mean to suggest a change - rather just an explicit check that its just hyperons on a handful of events... @civanch